plugins/tabletools/dialogs/tableCell.js

```diff
@@ -518,6 +412,78 @@ CKEDITOR.dialog.add( 'cellProperties', function( editor ) {
 		};
 	}
 
+	function getCellSizeFieldDefinition( fieldName ) {
+		return {
+			requiredContent: 'td{' + fieldName + '}',
+			type: 'hbox',
+			widths: [ '70%', '30%' ],
+			children: [ {
+				type: 'text',
+				id: fieldName,
+				width: '100px',
+				label: langCommon[ fieldName ],
+				validate: validate.number( langCell[ 'invalid' + CKEDITOR.tools.capitalize( fieldName ) ] ),
+
+				// Extra labelling of unit type.
+				onLoad: function() {
+					var unitType = this.getDialog().getContentElement( 'info', fieldName + 'Type' ),
+						labelElement = unitType.getElement(),
+						inputElement = this.getInputElement(),
+						ariaLabelledByAttr = inputElement.getAttribute( 'aria-labelledby' );
+
+					inputElement.setAttribute( 'aria-labelledby', [ ariaLabelledByAttr, labelElement.$.id ].join( ' ' ) );
+				},
+
+				setup: setupCells( function( element ) {
+					var attr = parseFloat( element.getAttribute( fieldName ), 10 ),
+						style = parseFloat( element.getStyle( fieldName ), 10 );
+
+					if ( !isNaN( style ) ) {
+						return style;
+					}
+					if ( !isNaN( attr ) ) {
+						return attr;
+					}
+
+					return;
+
+				} ),
+				commit: function( element ) {
+					var value = parseFloat( this.getValue(), 10 ),
+
+						// There might be no unit type, i.e. when multiple cells are
+						// selected but some of them have size expressed in pixels and some
+						// of them in percent. Try to re-read the unit from the cell in such
+						// case (https://dev.ckeditor.com/ticket/11439).
+						unit = this.getDialog().getValueOf( 'info', fieldName + 'Type' ) || getCellSizeUnitType( element, fieldName );
+
+					if ( !isNaN( value ) ) {
+						element.setStyle( fieldName, value + unit );
+					} else {
+						element.removeStyle( fieldName );
+					}
+
+					element.removeAttribute( fieldName );
+				},
+				'default': ''
+			}, {
+				type: 'select',
+				id: fieldName + 'Type',
+				label: editor.lang.table[ fieldName + 'Unit' ],
+				labelStyle: 'visibility:hidden;display:block;width:0;',
```

Comment:
There is `.cke_screen_reader_only` class inside `copyformatting/copyformatting.css`, it can be reused here. But it's probably a separate task.

Reply:
Reported in #3165.